src/Middleware/IpFilter.php

```diff
@@ -6,15 +6,16 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Validator\Rule\Ip;
 
 final class IpFilter implements MiddlewareInterface
 {
-    private $allowedIp;
+    private $ipValidator;
     private $responseFactory;
 
-    public function __construct(string $allowedIp, ResponseFactoryInterface $responseFactory)
+    public function __construct(Ip $ipValidator, ResponseFactoryInterface $responseFactory)
     {
-        $this->allowedIp = $allowedIp;
+        $this->ipValidator = (clone $ipValidator)->disallowSubnet()->disallowNegation();
```

Comment:
I think it's wrong.
Dependency must be already tuned before it will be passed in constructor or IP rule should be specified as `final`.

Reply:
Would you please elaborate?

Reply:
Dependency must be used as-is. 
What I shall do if I want to preconfigure IP rule dependency?
```php
IpFilter::class => function ($container) {
    return new IpFilter((new Ip())->allowNegation(), $container->get(ResponseFactoryInterface::class));
}
```
But this case it's no use, because a constructor do `disallowNegation()`

Reply:
Right. Worth considering. @kamarton?

Reply:
`dissallowNegation()` and `disallowSubnet()` refer to the value to be checked. `$_SERVER['REMOTE_ADDR']` cannot be `!127.0.0.1` or `8.8.0.0/16`.

Cloning can be removed, if `dissallowSubnet()` and `disallowNegation()` are called right before value validation.

Reply:
> Dependency must be already tuned before it will be passed in constructor or IP rule should be specified as final.

fixed.